capsules/src/alarm.rs

```diff
@@ -34,6 +40,7 @@ pub struct AlarmDriver<'a, A: Alarm<'a>> {
     num_armed: Cell<usize>,
     app_alarm: Grant<AlarmData>,
     prev: Cell<u32>,
+    min_tics: OptionalCell<usize>,
```

Comment:
I don't see this field ever containing anything else than an empty cell. Shouldn't we just use the `MIN_TICS_AT_16KHZ` constant until the new HIL solves this problem in a more principled manner?

Reply:
This is to avoid computing min_tics every time an alarm is set. It is computed lazy here `self.min_tics.set(min_tics)`:
````rust
let min_tics = self.min_tics.unwrap_or_else(|| {
	// scale the min tics from 16Khz to the actual frequency of the CPU
	let min_tics = (MIN_TICS_AT_16KHZ * (<A::Frequency>::frequency() as usize)) / 16000;
	self.min_tics.set(min_tics);
	min_tics
});
````

I made this computation at @alevy suggestion, as not every MCU has the same speed.

Reply:
If this is the concern, why not doing the computation in the `new` function? I think it would be clearer. That's some overhead if the command 4 is never used, but the computation happens only once at startup, so I think it's acceptable, given the clearer code that it allows. On the contrary, there's still some overhead in checking every time whether a cached value already exists.

This would also reduce memory overhead, because a single `usize` word would be needed to cache this value, as opposed to more memory to store the `OptionalCell<usize>`.

Also, in practice, aren't all the involved values known at compile time? In that case the compiler should be able to optimize the computation at compile time, and there would be no penalty due to the MCU's speed.

Reply:
That was my first idea, but the function is marked as const, so no non-const function calls can be made. <A::Frequency>::frequency() is defined in a trait, so it cannot be a const function.

I suspected that checking a enum is cheaper than two multiplications and a division.

Reply:
Indeed, there are the current limitations of `const` functions...

Reply:
I agree with @gendx that the new HIL will solve this.

For the time being, I'd advocate just avoiding the memoization. In practice, the computation _might_ be resolved at compiled time (since `MIN_TICS_AT_16KHZ` is constant, `A` is a specific implementation of `Alarm` that will in almost all cases have a constant implementation of `frequency()`.

And even if it's not, it's not a _particularly_ expensive computation.

If this were a permanent situation, that'd be one thing, but given we expect the new HIL to likely resolve this, it would be better not to have a temporary, and later extraneous, field in the driver's struct.